dspace-api/src/main/java/org/dspace/util/SolrImportExport.java

```diff
@@ -0,0 +1,583 @@
+				{
+					result.add(info.getName());
+				}
+			}
+		}
+		catch (IOException | SolrServerException e)
+		{
+			log.fatal("Cannot determine which fields are multi valued: " + e.getMessage(), e);
+		}
+		return result;
+	}
+
+	/**
+	 * Remove all documents from the Solr index with the given URL, then commit and optimise the index.
+	 *
+	 * @throws IOException if there is a problem in communicating with Solr.
+	 * @throws SolrServerException if there is a problem in communicating with Solr.
+	 * @param solrUrl URL of the Solr core to clear.
+	 */
+	public static void clearIndex(String solrUrl) throws IOException, SolrServerException
+	{
+		HttpSolrServer solr = new HttpSolrServer(solrUrl);
+		solr.deleteByQuery("*:*");
+		solr.commit();
+		solr.optimize();
+	}
+
+	/**
+	 * Exports documents from the given index to the specified target directory in batches of #ROWS_PER_FILE, starting at lastValue (or all documents).
+	 * See #makeExportFilename for the file names that are generated.
+	 *
+	 * @param indexName The index to export.
+	 * @param toDir The target directory for the export. Will be created if it doesn't exist yet. The directory must be writeable.
+	 * @param solrUrl The solr URL for the index to export. Must not be null.
+	 * @param timeField The time field to use for sorting the export. Must not be null.
+	 * @param lastValue Optionally, from when to export. See options for allowed values. If null or empty, all documents will be exported.
+	 * @throws SolrServerException if there is a problem with exporting the index.
+	 * @throws IOException if there is a problem creating the files or communicating with Solr.
+	 * @throws SolrImportExportException if there is a problem in communicating with Solr.
+	 */
+	public static void exportIndex(String indexName, String toDir, String solrUrl, String timeField, String lastValue) throws SolrServerException, IOException, SolrImportExportException
+	{
+		if (StringUtils.isBlank(solrUrl))
+		{
+			throw new SolrImportExportException("Could not construct solr URL for index" + indexName + ", aborting export.");
+		}
+
+		File targetDir = new File(toDir);
+		if (!targetDir.exists())
+		{
+			//noinspection ResultOfMethodCallIgnored
+			targetDir.mkdirs();
+		}
+		if (!targetDir.exists())
+		{
+			throw new SolrImportExportException("Could not create target directory " + toDir + ", aborting export of index ");
+		}
+
+		SolrQuery query = new SolrQuery("*:*");
+		if (StringUtils.isNotBlank(lastValue))
+		{
+			String lastValueFilter = makeFilterQuery(timeField, lastValue);
+			if (StringUtils.isNotBlank(lastValueFilter))
+			{
+				query.addFilterQuery(lastValueFilter);
+			}
+		}
+
+		HttpSolrServer solr = new HttpSolrServer(solrUrl);
+		SolrDocumentList results = solr.query(query).getResults();
+		long totalRecords = results.getNumFound();
+
+		query.setRows(ROWS_PER_FILE);
+		query.set("wt", "csv");
+		query.set("fl", "*");
+		query.setSort(timeField, SolrQuery.ORDER.asc);
```

Comment:
The sorting appears to cause the memory issues that people have reported running into when doing the export. I ran the Eclipse Memory Analyzer on a heap dump that @hardyoyo provided, and both problem suspects reported by that tool point to something related to sorting Solr documents. I'm going to try and change the code to process documents in weekly or monthly batches (using filter queries) so that it only has to sort the documents in a given batch.